tests/test_frontend_testing_api/test_reservation.py

```diff
@@ -0,0 +1,151 @@
+    assert not user.permissions.has_general_role()
+    assert not user.permissions.has_role_for_units_or_their_unit_groups(units=Unit.objects.all())
+
+
+def test_frontend_api__reservation__set_datetime__default_options(api_client):
+    data = {
+        "current_datetime": "2022-01-01T12:00:00",
+        "user": {},
+        "params": {},
+    }
+    response = api_client.post(reverse(_ENDPOINT), data=data, format="json")
+    assert response.status_code == 201, response.json()
+
+    reservation_unit = ReservationUnit.objects.first()
+    assert reservation_unit.is_archived is False
+    assert reservation_unit.is_draft is False
+    assert reservation_unit.require_reservation_handling is False
+    assert reservation_unit.cancellation_rule.can_be_cancelled_time_before is None
+
+    reservation = Reservation.objects.first()
+    assert reservation.begin == local_datetime(2022, 1, 2, 12)
+    assert reservation.state == ReservationStateChoice.CONFIRMED
+    assert reservation.recurring_reservation is None
+
+    pricing = ReservationUnitPricing.objects.first()
+    assert pricing.highest_price == 0
+
+    assert PaymentOrder.objects.count() == 0
+
+    stop_time_freeze()
+
+
+def test_frontend_api__reservation__set_datetime__non_default_options(api_client):
+    data = {
+        "current_datetime": "2030-01-01T00:00:00",
+        "user": {},
+        "params": {
+            "state": ReservationStateChoice.REQUIRES_HANDLING,
+            "is_archived": True,
+            "is_draft": True,
+            "is_cancellable": True,
+            "is_paid_reservation": True,
+            "is_paid_on_site": True,
+            "is_past": True,
+            "is_part_of_series": True,
+        },
+    }
+    response = api_client.post(reverse(_ENDPOINT), data=data, format="json")
+    assert response.status_code == 201, response.json()
+
+    assert ReservationUnit.objects.count() == 1
+    reservation_unit = ReservationUnit.objects.first()
+    assert reservation_unit.is_archived is True
+    assert reservation_unit.is_draft is True
+    assert reservation_unit.require_reservation_handling is True
+    assert reservation_unit.cancellation_rule.can_be_cancelled_time_before == datetime.timedelta(hours=24)
+
+    assert Reservation.objects.count() == 1
+    reservation = Reservation.objects.first()
+    assert reservation.begin == local_datetime(2029, 12, 31, 00)
+    assert reservation.state == ReservationStateChoice.REQUIRES_HANDLING
+    assert reservation.recurring_reservation is not None
+
+    assert ReservationUnitPricing.objects.count() == 1
+    pricing = ReservationUnitPricing.objects.first()
+    assert pricing.highest_price == 10
+
+    assert PaymentOrder.objects.count() == 1
+    payment_order = PaymentOrder.objects.first()
+    assert payment_order.payment_type == PaymentType.ON_SITE
+    assert payment_order.status == OrderStatus.DRAFT
+
+    stop_time_freeze()
```

Comment:
See comment: https://github.com/City-of-Helsinki/tilavarauspalvelu-core/pull/1511/files#r1933342361